r/src/arrow_cpp11.h

```diff
@@ -157,19 +157,42 @@ struct ns {
   static SEXP arrow;
 };
 
+// Specialize this struct to define a default value to be used when NULL is given.
```

Comment:
Why is this change necessary?

Reply:
It's a convenience for passing optional parameters; instead of writing multiple overloads of a method with different permutations of optional parameters specified those parameters can be set to `NULL` in R. It's not necessary; should I revert it?